x-pack/plugins/files/server/routes/find.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema, TypeOf } from '@kbn/config-schema';
+import type { Ensure } from '@kbn/utility-types';
+import type { FilesRouter } from './types';
+
+import { FindFilesHttpEndpoint, FILES_API_ROUTES } from '../../common/api_routes';
+import type { FilesRequestHandler } from './types';
+
+export const method = 'post' as const;
+
+const stringOrArrayOfStrings = schema.oneOf([schema.string(), schema.arrayOf(schema.string())]);
+
+const bodySchema = schema.object({
+  kind: schema.maybe(stringOrArrayOfStrings),
+  name: schema.maybe(stringOrArrayOfStrings),
+  mimeType: schema.maybe(stringOrArrayOfStrings),
+  status: schema.maybe(stringOrArrayOfStrings),
+  kuery: schema.maybe(stringOrArrayOfStrings),
+  meta: schema.maybe(schema.object({}, { unknowns: 'allow' })),
```

Comment:
The contents of `meta` is currently not validate, maybe we can set some restrictions, like no nested objects, etc..

Reply:
I think it could still be useful to allow nested objects in meta given we do not know all possible use cases or preferences.

Aside: I realised that dynamic objects will not be possible with saved objects (it is an explicit limitation). After speaking with Rudolf I realised that `flattened` would be a good fit for this field since we expect all leaf-values to be `keywords`.

Given that I think it will still be nice to allow nested objects (internally they are flattened anyway). I don't see a technical reason they should not be able to do it. WDYT?